tasks/utils.js

```diff
@@ -50,3 +51,39 @@ module.exports.spawnablePath = function (path) {
     }
     return path;
 };
+
+
+// return BrowserSync URL root
+var getBrowserSyncRootUrl = module.exports.getBrowserSyncRootUrl = function (options) {
+    return 'http://localhost:' + options.get('port');
+};
+
+// return BrowserSync JS client URL
+module.exports.getBrowserSyncClientUrl = function (options) {
+    var connectUtils = require('browser-sync/lib/connect-utils');
+    return getBrowserSyncRootUrl(options) + connectUtils.clientScript(options);
+};
+
+// Setup BrowserSync and catch errors gracefully
+module.exports.initBrowserSync = function (bs_opts, ready) {
+    var bs = require("browser-sync").create();
+    var opts = extend({
+        ui: false,
+        port: 35829,
+        ghostMode: false,
+        open: false,
+        notify: false,
+        logSnippet: false,
+        socket: {domain: getBrowserSyncRootUrl}
+    }, bs_opts);
+    var err;
+    try {
+        bs.init(opts, function (){
+            ready( err ? null : bs);
```

Comment:
This is rather strange use of error and I couldn't image this code actually works ;)
if `bs.init` throws how callback function can be invoked at all?